server/routes/searchComments.ts

```diff
@@ -0,0 +1,98 @@
+/* eslint-disable quotes */
+import { Request, Response, NextFunction } from 'express';
+import { QueryTypes } from 'sequelize';
+import lookupCommunityIsVisibleToUser from '../util/lookupCommunityIsVisibleToUser';
+import { DB } from '../database';
+
+const Errors = {
+  UnexpectedError: 'Unexpected error',
+  QueryMissing: 'Must enter query to begin searching',
+  QueryTooShort: 'Query must be at least 4 characters',
+  NoCommunity: 'Title search must be community scoped'
+};
+
+const searchComments = async (
+  models: DB,
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  let replacements = {};
+
+  if (!req.query.search) {
+    return next(new Error(Errors.QueryMissing));
+  }
+  if (req.query.search.length < 4) {
+    return next(new Error(Errors.QueryTooShort));
+  }
+
+  // Community-scoped search
+  let communityOptions = '';
+  if (req.query.chain || req.query.community) {
+    const [chain, community, error] = await lookupCommunityIsVisibleToUser(
+      models,
+      req.query,
+      req.user
+    );
+    if (error) return next(new Error(error));
+
+    // set up query parameters
+    communityOptions = community
+      ? `AND "OffchainComments".community = :community`
+      : `AND "OffchainComments".chain = :chain `;
+    replacements = community
+      ? { community: community.id }
+      : { chain: chain.id };
+  }
+
+  const sort = req.query.sort === 'Newest'
+    ? 'ORDER BY "OffchainComments".created_at DESC'
+    : req.query.sort === 'Oldest'
+    ? 'ORDER BY "OffchainComments".created_at ASC'
+    : 'ORDER BY rank DESC'
+
+  replacements['searchTerm'] = req.query.search;
+  replacements['limit'] = 50; // must be same as SEARCH_PAGE_SIZE on frontend
+
+  // query for both threads and comments, and then execute a union and keep only the most recent :limit
+  let comments;
+  try {
+    comments = await models.sequelize.query(
+      `
+  SELECT
+      "OffchainThreads".title,
+      "OffchainComments".text,
+      "OffchainComments".root_id as proposalId,
+      'comment' as type,
+      "Addresses".id as address_id,
+      "Addresses".address,
+      "Addresses".chain as address_chain,
+      "OffchainComments".created_at,
+      "OffchainThreads".chain,
+      "OffchainThreads".community,
+      ts_rank_cd("OffchainComments"._search, query) as rank
+    FROM "OffchainComments"
+    JOIN "OffchainThreads" ON "OffchainThreads".id =
+        CASE WHEN root_id ~ '^discussion_[0-9\\.]+$' THEN CAST(REPLACE(root_id, 'discussion_', '') AS int) ELSE NULL END
+    JOIN "Addresses" ON "OffchainComments".address_id = "Addresses".id, 
+    websearch_to_tsquery('english', :searchTerm) as query
+    WHERE query @@ "OffchainComments"._search ${communityOptions} 
+    ${sort} LIMIT :limit
+`,
+      {
+        replacements,
```

Comment:
Quick fix but we should use bind here instead of replacements, to keep with all the other routes.